Analysis/src/Error.cpp

```diff
@@ -112,21 +112,22 @@ struct ErrorConverter
 
     std::string operator()(const Luau::CountMismatch& e) const
     {
+        const std::string expectedS = e.expected == 1 ? "" : "s";
+        const std::string actualS = e.actual == 1 ? "" : "s";
+        const std::string actualVerb = e.actual == 1 ? "is" : "are";
+
         switch (e.context)
         {
         case CountMismatch::Return:
-        {
-            const std::string expectedS = e.expected == 1 ? "" : "s";
-            const std::string actualS = e.actual == 1 ? "is" : "are";
-            return "Expected to return " + std::to_string(e.expected) + " value" + expectedS + ", but " + std::to_string(e.actual) + " " + actualS +
-                   " returned here";
-        }
+            return "Expected to return " + std::to_string(e.expected) + " value" + expectedS + ", but " +
+                   std::to_string(e.actual) + " " + actualVerb + " returned here";
         case CountMismatch::Result:
             if (e.expected > e.actual)
-                return "Function returns " + std::to_string(e.expected) + " values but there are only " + std::to_string(e.expected) +
-                       " values to unpack them into.";
+                return "Function returns " + std::to_string(e.expected) + " values, but there " + actualVerb +
```

Comment:
The new test case shows that this branch is not currently reachable. My guess would be that `tryUnifyVariadics` does not complain if there are less symbols than returned values.

What do we do? Allow ignoring excess return values and remove this branch?

From a Lua perspective, it is common practice to ignore excess return values. For example:
```lua
table.remove(t) -- would return the removed item, but we aren't assigning it to any symbols
local point = game.Workspace.CurrentCamera:WorldToScreenPoint() -- I don't care if the point is in bounds of the screen, but Luau would complain unless I stored it in a dummy symbol "_".
```